src/mainboard/amd/romecrb/src/acpi.rs

```diff
@@ -11,44 +11,51 @@ use util::round_up_4k;
 /// Total number of bytes occupied by the BIOS tables is returned.
 pub fn setup_acpi_tables(w: &mut impl core::fmt::Write, start: usize, cores: u32) -> usize {
     // calculate offsets first
-    // variables with suffix `_offset` mean the offset in `low_mem`. They are
-    // also the guest physical address of the corresponding tables, since `low_mem` is
-    // mapped to the lowest memory of the guest's physical address space.
+    // variables with suffix `_offset` mean the offset in `low_mem`.
+
+    const NUM_XSDT_ENTRIES: usize = 4;
 
     let rsdp_offset = start;
     let xsdt_offset = rsdp_offset + size_of::<AcpiTableRsdp>();
     let xsdt_entry_offset = xsdt_offset + size_of::<AcpiTableHeader>();
-    const NUM_XSDT_ENTRIES: usize = 8;
     let fadt_offset = xsdt_entry_offset + NUM_XSDT_ENTRIES * size_of::<u64>();
     let facs_offset = fadt_offset + size_of::<AcpiTableFadt>();
     let dsdt_offset = facs_offset + size_of::<AcpiTableFacs>();
+
+    // let madt_offset = dsdt_offset + 69887;
     let madt_offset = dsdt_offset + DSDT_DSDTTBL_HEADER.len();
     let madt_local_apic_offset = madt_offset + size_of::<AcpiTableMadt>();
-    let io_apic_offset = madt_local_apic_offset + cores as usize * size_of::<AcpiMadtLocalApic>();
-    let local_x2apic_offset = io_apic_offset + size_of::<AcpiMadtIoApic>();
-    let local_lapicnmi_offset = local_x2apic_offset + cores as usize * size_of::<AcpiMadtLocalX2apic>();
-    let local_isor_offset = local_lapicnmi_offset + size_of::<AcpiMadtLocalX2ApicNMI>();
-    let mcfg_offset = local_isor_offset + 2 * size_of::<AcpiMadtInterruptOverride>();
-    let total_size = mcfg_offset + size_of::<AcpiTableMcfg>() - start;
+    let madt_local_x2apic_offset = madt_local_apic_offset + cores as usize * size_of::<AcpiMadtLocalApic>();
```

Comment:
Could a crate like [memoffset](https://docs.rs/memoffset/0.6.1/memoffset/macro.offset_of.html) be useful here?

Reply:
Probably much better would be to generate an SVD file, then use [svd2rust](https://docs.rs/svd2rust/0.17.0/svd2rust/) to generate register descriptions for us, but this is probably a fair bit of tedious work, then a major refactor.

Reply:
I tihnk the day is coming when we rework the way we do this, and hopefully in a way we can take it back to xhype.